MREUnityRuntime/MREUnityRuntimeLib/App/MixedRealityExtensionApp.cs

```diff
@@ -577,8 +579,21 @@ private void OnCreateFromPrefab(CreateFromPrefab payload, Action onCompleteCallb
         {
             try
             {
-                var createdActors = _assetLoader.CreateFromPrefab(payload.PrefabId, payload.Actor?.ParentId);
-                ProcessCreatedActors(payload, createdActors, onCompleteCallback);
+                var curGeneration = generation;
+                MREAPI.AppsAPI.AssetCache.OnCached(payload.PrefabId, prefab =>
+                {
+                    if (this == null || _appState != AppState.Running || generation != curGeneration) return;
```

Comment:
>_appState != AppState.Running [](start = 40, length = 29)

_appState != AppState.Running isn't the right check to do here. State synchronization happens before the app enters the running state, and assets are loaded during state synchronization.

Reply:
I think _conn.IsActive would be a safer alternative.

---
In reply to: [307544285](https://github.com/microsoft/mixed-reality-extension-unity/pull/92#discussion_r307544285) [](ancestors = 307544285)

Reply:
Fixed

---
In reply to: [307544541](https://github.com/microsoft/mixed-reality-extension-unity/pull/92#discussion_r307544541) [](ancestors = 307544541,307544285)